src/Utils/request/errorHandler.ts

```diff
@@ -76,6 +82,24 @@ type PydanticError = {
   url: string;
 };
 
+function isStructuredError(cause: HTTPError["cause"]): boolean {
+  return typeof cause === "object" && !Array.isArray(cause);
+}
+
+function handleStructuredErrors(cause: HTTPError["cause"]) {
+  if (cause && typeof cause === "object" && !Array.isArray(cause)) {
+    Object.values(cause).forEach((value) => {
+      if (Array.isArray(value)) {
+        value.forEach((err) => {
+          toast.error(`${err}`);
+        });
+      } else if (typeof value === "string") {
+        toast.error(value);
+      }
+    });
+  }
+}
```

Comment:
this check is already done by `isStructuredError`. why check again inside the handler?

you can make the `cause` type of handler more restrictive to `Record<string, string | string[]>`

and update the return type of isStructeredError to `cause is Record<string, string | string[]>` instead.


```suggestion
function isStructuredError(cause: HTTPError["cause"]): cause is Record<string, string | string[]> {
  return typeof cause === "object" && !Array.isArray(cause);
}

function handleStructuredErrors(cause: Record<string, string | string[]>) {
  for (const value of Object.values(cause)) {
    if (Array.isArray(value)) {
      value.forEach((err) => toast.error(err));
      return;
    }
    if (typeof value === "string") {
      toast.error(value);
      return;
    }
  }
}
```